reco_utils/dataset/split_utils.py

```diff
@@ -159,4 +159,8 @@ def split_pandas_data_with_ratios(data, ratios, seed=1234, resample=False):
 
     splits = np.split(data, [round(x * len(data)) for x in split_index])
 
+    # Add split index (this makes splitting by group more efficient).
```

Comment:
The returned result is a list of numpy arrays, isn't it? The docstring says dataframes.
Also it would be good to mention in the docstring the 'split_index' attribute.

Reply:
I checked and it returns dataframes. 

@yueguoguo looking at this function, I see that there is no control of the input ratios. For example, `ddd = split_pandas_data_with_ratios(df, [0.6, 0.2, 0.5])` will work, it will split 60%, 20% and 20%. The function is created so the last figure is always ignored. 

I'll send a PR to fix this

Reply:
https://github.com/Microsoft/Recommenders/pull/533

Reply:
This is interesting because the numpy docs say that it should return a list of ndarrays 
https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.split.html
But as you said, it returns a list of dataframes if data is a dataframe and a list of ndarrays if data is an ndarray.

About the ratios, the functions in `python_splitters.py` expect them to be renormalized if they don't add up to one.